app/src/main/java/com/kabouzeid/gramophone/appwidgets/AppWidgetClassic.java

```diff
@@ -93,26 +94,41 @@ public void run() {
+                        .into(new SimpleTarget<BitmapPaletteWrapper>(imageSize, imageSize) {
                             @Override
-                            public void onResourceReady(Bitmap resource, GlideAnimation<? super Bitmap> glideAnimation) {
-                                update(resource);
+                            public void onResourceReady(BitmapPaletteWrapper resource, GlideAnimation<? super BitmapPaletteWrapper> glideAnimation) {
+                                Palette palette = resource.getPalette();
+                                update(resource.getBitmap(), palette.getVibrantColor(palette.getMutedColor(MaterialValueHelper.getSecondaryTextColor(appContext, true))));
                             }
 
                             @Override
                             public void onLoadFailed(Exception e, Drawable errorDrawable) {
                                 super.onLoadFailed(e, errorDrawable);
-                                update(null);
+                                update(null, MaterialValueHelper.getSecondaryTextColor(appContext, true));
                             }
 
-                            private void update(@Nullable Bitmap bitmap) {
+                            private void update(@Nullable Bitmap bitmap,int color) {
                                 appWidgetView.setViewVisibility(R.id.image, View.VISIBLE);
+
+                                // Set correct drawable for pause state
+                                int playPauseRes = isPlaying ? R.drawable.ic_pause_white_24dp : R.drawable.ic_play_arrow_white_24dp;
+                                appWidgetView.setImageViewBitmap(R.id.button_toggle_play_pause, createBitmap(Util.getTintedVectorDrawable(service, playPauseRes, color), 1f));
+
+                                // Set prev/next button drawables
+                                appWidgetView.setImageViewBitmap(R.id.button_next, createBitmap(Util.getTintedVectorDrawable(service, R.drawable.ic_skip_next_white_24dp, color), 1f));
+                                appWidgetView.setImageViewBitmap(R.id.button_prev, createBitmap(Util.getTintedVectorDrawable(service, R.drawable.ic_skip_previous_white_24dp, color), 1f));
+
+                                Drawable image;
+
                                 if (bitmap == null) {
-                                    appWidgetView.setImageViewResource(R.id.image, R.drawable.default_album_art);
+                                    image = service.getResources().getDrawable(R.drawable.default_album_art);
                                 } else {
-                                    appWidgetView.setImageViewBitmap(R.id.image, bitmap);
+                                    image = new BitmapDrawable(bitmap);
                                 }
+
+                                appWidgetView.setImageViewBitmap(R.id.image, createRoundedBitmap(image, imageSize, imageSize, cardRadius, 0, cardRadius, 0));
```

Comment:
I don't think this actually seems to be working correctly here. I see black pixels at the top left corner, and the bottom left corner looks normal (i.e. not rounded, and no black pixels as if it was rounded). Could be something to do with the layout?

I also noticed that the card and classic widgets stretch the album art to fit the square if they're not already square, while the small widget fits it with black bars around it.